packages/myst-frontmatter/src/frontmatter/validators.ts

```diff
@@ -667,21 +667,39 @@ export function validateBinderHubOptions(input: any, opts: ValidationOptions) {
   if (defined(value.url)) {
     output.url = validateUrl(value.url, incrementOptions('url', opts));
   }
-  if (defined(value.ref)) {
-    output.ref = validateString(value.ref, incrementOptions('ref', opts));
-  }
-  if (defined(value.repo)) {
-    output.repo = validateString(value.repo, {
-      ...incrementOptions('repo', opts),
-      regex: GITHUB_USERNAME_REPO_REGEX,
-    });
-  }
   if (defined(value.provider)) {
-    output.provider = validateEnum<BinderProviders>(value.provider, {
+    output.provider = validateString(value.provider, {
       ...incrementOptions('provider', opts),
-      enum: BinderProviders,
+      regex: /.+/,
     });
+    if (output.provider?.match(/^(git|github|gitlab|gist)$/i)) {
+      if (defined(value.repo)) {
+        output.repo = validateString(value.repo, {
+          ...incrementOptions('repo', opts),
+          regex: GITHUB_USERNAME_REPO_REGEX,
+        });
+      }
+    } else {
+      // otherwise provider can be any value, validate as any non empty string
+      output.repo = validateString(value.repo, {
+        ...incrementOptions('repo', opts),
+        regex: /.+/,
+      });
+    }
+  } else {
+    // assumes a WellKnownRepoProvider defaulting to 'github', validate repo as a git*/gist spec
+    if (defined(value.repo)) {
```

Comment:
These six lines are duplicated above. Maybe another place we can rearrange logic and combine if statements... like
```
if (defined(value.provider)) {
  // validate any string for provider
}
if (defined(value.repo)) {
  if (output.provider && !output.provider.match(...) {
    // validate any string for repo
  } else {
    // validate github for repo
  }
}
```

I think the only change to the logic with this addresses my previous comment around undefined `repo`; not sure if that's right or wrong.

Reply:
did something like this